traefik/templates/service.yaml

```diff
@@ -3,19 +3,22 @@
 {{ $tcpPorts := dict }}
 {{ $udpPorts := dict }}
 {{- range $name, $config := .Values.ports }}
-  {{- if eq (toString $config.protocol) "UDP" }}
-    {{ $_ := set $udpPorts $name $config }}
-  {{- else }}
-    {{ $_ := set $tcpPorts $name $config }}
+  {{- if $config.expose }}
+    {{- if eq (toString $config.protocol) "UDP" }}
+      {{ $_ := set $udpPorts $name $config }}
+    {{- else }}
+      {{ $_ := set $tcpPorts $name $config }}
+    {{- end }}
   {{- end }}
 {{- end }}
 
+{{- if or (gt (len $tcpPorts) 0) (gt (len $udpPorts) 0) }}
 apiVersion: v1
 kind: List
```

Comment:
Same here, would you please explain what is this List for?

Reply:
A List can include one or more objects, with no distinction on type -- we could concatenate most templates into a single file.

Another way to write it would have been to add separators (`---`) between objects, though I'm not sure unit tests would be able to check specific objects, as we currently do (`items[$index].(spec|metadata|...)`).

Arguably, we could have used separate templates for UDP and TCP services: there's not a lot of shared code, which could have been a reason for using a List.
Maybe there's another explanation I'm missing?
Otherwise I can split them and fix tests going with one file per Service.

Reply:
Thanks for explaining that. So, we need to just rebase the PR and then we can merge it.